Fix/td 15/ordered task manager

chuRealOff opened the pull request:
Прошу утвердить изменения

chuRealOff committed:
Refactor allTasks() func

- naming has been changed
- test assertions logic experienced a few changes

chuRealOff committed:
Refactor allTasks() func

- naming has been changed
- test assertions logic experienced a few changes

chuRealOff committed:
Change test functions naming

The following functions names were changed:
- test_ifSorted_byCompletion() to test_completedTasks_shouldReturnCompletedTasks()
- test_ifSorted_byTasksInProgress() to test_uncompletedTasks_shouldReturnUncompletedTasks()

chuRealOff committed:
Change test functions naming

The following functions names were changed:
- test_ifSorted_byCompletion() to test_completedTasks_shouldReturnCompletedTasks()
- test_ifSorted_byTasksInProgress() to test_uncompletedTasks_shouldReturnUncompletedTasks()

djud17 committed:
TD-14 Fixed taskManager tests (#32)

* Fixed taskManager tests

* Fix test names

djud17 commented on TaskManagerPackage/Tests/TaskManagerPackageTests/OrderedTaskManagerTests.swift:
```diff
@@ -62,22 +62,24 @@ final class OrderedTaskManagerTests: XCTestCase {
 		)
 	}
```
1. Поправь названия тестов выше
2. По гайдлайну не должно быть setUp() и tearDown() в unit-тестах.
Все настройки выносятся в private extension в метод makeSut()

По идее тебе здесь нужно протестировать только отличия от обычного taskManager - то есть сортировку. Ну разве что можно протестировать по аналогии с SectionManager, что orderedTaskManager вызывает нужные методы у обычного taskManager.

chuRealOff replied:
Пофиксил

chuRealOff committed:
Final refactoring has been performed

- makeSut() -> OrderedTaskManager func has been added to the project
- functions naming changed in accordance with SSG
- more assert text annotations added
- minor mark down issues have been fixed

chuRealOff committed:
Merge branch 'fix/TD-15/OrderedTaskManager' of https://github.com/djud17-org/todoList into fix/TD-15/OrderedTaskManager

# Conflicts:
#	TaskManagerPackage/Tests/TaskManagerPackageTests/OrderedTaskManagerTests.swift

djud17 commented on TaskManagerPackage/Tests/TaskManagerPackageTests/OrderedTaskManagerTests.swift:
```diff
@@ -12,79 +12,89 @@ final class OrderedTaskManagerTests: XCTestCase {
+		XCTAssertEqual(tasks.count, 2, "Количество задание не соответствует исходному значению")
+		XCTAssertTrue(tasks[0].taskStatus == .completed, "Задания не отсоритрованы по завершенным")
+		XCTAssertTrue(tasks[1].taskStatus == .planned, "Задания должны быть отсортированы по завершенным")
 	}
```
Используй корректные ассерты - XCTAssertEqual

chuRealOff replied:
исправил

djud17 commented on TaskManagerPackage/Tests/TaskManagerPackageTests/TaskManagerTests.swift:
Как мой файл попал?)

chuRealOff replied:
для меня это абсолютная загадка )

chuRealOff committed:
XCTAssertTrue replaced by XCTAssertEqual in a pair of lines
